### ISSUE TITLE
DetailsList with Narrator + Edge when using Browse mode does not read row info.

### ISSUE THREAD

**Issue by mibadget** [OfficeDev/office-ui-fabric-react#10268]
### Environment Information
URL: https://developer.microsoft.com/en-us/fabric#/controls/web/detailslist
Browser: Edge 
OS: Windows Version 10.0.18362 Build 18362

### Describe the issue:
Using the Details List component, when focus is on a row in the details list, no information is read by narrator pertaining to the row information such as which row the focus is on. Sighted users can see screenshot below with a narrator text output.

### Please provide a reproduction of the issue in a codepen:
This reproduces on all instances of the Deatails list in the Official Office UI Fabric page. See the environment URL at the top. each Details list example has an export to CodePen example.

![image](https://user-images.githubusercontent.com/53191940/63711773-0d8e8300-c7f1-11e9-83f9-78bde012ddd9.png)

#### Actual behavior:
The Screen Reader reads all of the information contained in the row but no information about which row the user is on.

#### Expected behavior:
The screen reader should read All of the row's children information then read "Row n of m" or something similar.

### Documentation describing expected behavior
WAI-ARIA specifically calls out that elements with role="row" should have an aria-rowindex and the grid should have aria-rowcount . This is so that Row information is available to focusable elements in the grid for screen readers to communicate to the user where they are in the grid.

https://www.w3.org/TR/wai-aria-1.1/#aria-rowindex
[MAS 1.3.1](https://microsoft.sharepoint.com/:w:/r/teams/msenable/_layouts/15/WopiFrame.aspx?sourcedoc=%7b54f28d1f-a2d1-4dcd-84e1-5c9b87e8aba4%7d)

**Comment by smhigley** [comment]
I took a quick look at this, and I think it may root cause in DetailsList putting focus on `<tr>` elements instead of cells. This isn't technically allowed (https://www.w3.org/TR/wai-aria-1.1/#grid), and I verified that Narrator + Edge reads the row index correctly when using the usual scan mode table nav shortcuts.

**Comment by maxwellred** [comment]
Hi @smhigley thanks for taking a look and commenting your findings! I was only able to get the first row (DetailsList Header) to read out the position in Narrator scan mode. I was able to find a way to get subsequent rows to read out by using aria-posinset and aria-setsize on each row. @mibadget I mocked the examples given [here](https://www.w3.org/TR/wai-aria-1.1/#aria-rowindex) (Examples 30 and 31) but Narrator does not read out any row info so this approach might be a Narrator issue. I feel the best course of action is to have the method I describe using the aria-posinset and aria-setsize, what are your thoughts on this?

**Comment by dzearing** [comment]
@ThomasMichon @cschlechty @betrue-final-final We would appreciate your assistance triaging this issue.

@smhigley I couldn't quite find in the spec you posted where it mentions role should not receive focus. Could you paste the part you're looking at?
